docs/DeveloperGuide.md

```diff
@@ -1,34 +1,938 @@
+(Jia Ern)
+
+The revise feature allows the user to start a revision on a chapter and can only be done when the user is in the module level. 
+
+#### 4.4.1. Implementation
+The revise mechanism is facilitated by `ReviseCommand`. It extends from the abstract class `Command`. 
+
+In addition, it implements the following operations:
+* `ReviseCommand#execute()` — oversees the entire revise process and calls the respective methods when necessary.
+* `ReviseCommand#getChapter()` — gets `chapter` based on the index the user provided.
+* `ReviseCommand#getCards()` — gets a list of `card` in `chapter`.
+* `ReviseCommand#promptNotDue()` — prompts user if he still wants to revise a `chapter` that is not due.
+* `ReviseCommand#reviseCard()` — makes use of `ui` to show the contents of each card to the user and gets rating input for the particular card.
+* `ReviseCommand#rateCard()` — gets user input on difficulty of a flashcard.
+* `ReviseCommand#repeatRevision()` — repeats revision for cards which user could not answer. 
+
+For instance, the user wants to start a revision for `Chapter 1` in the module `CS2113T`, a detailed description of what happens is shown below:
+
+Step 1: The user is currently in `CS2113T` at the module level.
+
+Step 2: The user enters `revise 1` command to revise the first chapter in the module — which in this case is `Chapter 1`. The `revise` command creates `ReviseCommand` which will then be executed.
+
+Step 3: `ReviseCommand#execute` gets `Chapter1` based on the index provided as well as a list of `card` under the particular chapter by calling `ReviseCommand#getChapter` and `ReviseCommand#getCards` respectively.
+
+Step 4: If the `chapter` is not due for revision yet, `ReviseCommand#promptNotDue()` will prompt the user if he still wants to revise a `chapter` that is not due. If the user enters `Y`, the Revise feature will continue execution, else it will return to the main class Kaji, and wait for the next command.
+
+Step 5: `ReviseCommand#execute` loops through each flashcard and shows the user its contents by calling `ReviseCommand#reviseCard()`  and inside it, `ReviseCommand#rateCard()` is called and makes use of `Ui#getUserInput()` to get user to rate the difficulty of each flashcard. 
+
+Step 6: A success message of completeing the revision will be shown to the user through `Ui#showToUser()`.
+
+Step 7: `Scheduler#computeDeckDeadline()` then calculates the new deadline for the `chapter` and passes the result to `Chapter#setueBy()` to set the new deadline for the `chapter`.
+
+Step 6: `ReviseCommand#repeatRevision` then repeats the revision session on cards which the user could not answer.
+
+Step 7: Lastly, `ReviseCommand#execute()` calls `HistoryCommand#addHistory()` to keep a record of the chapter revised so that the user can review what revision has been completed.
+
+The following sequence diagram shows how the revise feature works:
+![Sequence Diagram of Revise](DG_Images/revise_seq_diagram.png)
```

Comment:
The diagram used is rather complicated and quite difficult to read given its small font and compact information. Would it be better if you consider splitting it into different diagrams, or omit some of the less important detials?